core/app/models/spree/stock_location.rb

```diff
@@ -108,10 +108,7 @@ def create_stock_items
 
       def ensure_one_default
         if self.default
-          StockLocation.where(default: true).where.not(id: self.id).each do |stock_location|
-            stock_location.default = false
-            stock_location.save!
-          end
+          StockLocation.where(default: true).where.not(id: self.id).update_all(default: false)
```

Comment:
update_all creates *nasty* SQL deadlocks that have been plaguing Spree for years. Although it *looks* less clean, looping actually eliminates the chances of SQL deadlocks. 

You would only see or experience SQL deadlocks on larger stores, but trust me, they are created by update_all and they are really, really bad.